src/Oro/Bundle/PimDataGridBundle/spec/Validator/Constraints/UniqueDatagridViewEntityValidatorSpec.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace spec\Oro\Bundle\PimDataGridBundle\Validator\Constraints;
+
+use Akeneo\UserManagement\Component\Model\User;
+use Akeneo\UserManagement\Component\Model\UserInterface;
+use Oro\Bundle\PimDataGridBundle\Entity\DatagridView;
+use Oro\Bundle\PimDataGridBundle\Repository\DatagridViewRepositoryInterface;
+use Oro\Bundle\PimDataGridBundle\Validator\Constraints\UniqueDatagridViewEntity;
+use Oro\Bundle\PimDataGridBundle\Validator\Constraints\UniqueDatagridViewEntityValidator;
+use PhpSpec\ObjectBehavior;
+use Symfony\Component\Validator\Context\ExecutionContextInterface;
+use Symfony\Component\Validator\Violation\ConstraintViolationBuilderInterface;
+
+class UniqueDatagridViewEntityValidatorSpec extends ObjectBehavior
+{
+    function let(
+        ExecutionContextInterface $context,
+        DatagridViewRepositoryInterface $datagridViewRepository
+    ) {
+        $this->beConstructedWith($datagridViewRepository);
+
+        $this->initialize($context);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(UniqueDatagridViewEntityValidator::class);
+    }
+
+    function it_adds_violation_to_the_context_if_a_public_datagrid_view_already_exists_with_the_same_label(
+        ExecutionContextInterface $context,
+        DatagridViewRepositoryInterface $datagridViewRepository,
+        ConstraintViolationBuilderInterface $constraintViolationBuilder
+    ) {
+        $constraint = new UniqueDatagridViewEntity();
+
+        $datagridView = new DatagridView();
+        $datagridView->setId('1');
+        $datagridView->setType(DatagridView::TYPE_PUBLIC);
+        $datagridView->setLabel('The best public view');
+
+        $datagridViewInDatabase = clone $datagridView;
+        $datagridViewInDatabase->setId('2');
+
+        $datagridViewRepository->findPublicDatagridViewByLabel($datagridView->getLabel())
+            ->willReturn($datagridViewInDatabase);
+
+        $context->buildViolation('The same label is already set on another view')
+            ->willReturn($constraintViolationBuilder);
+        $constraintViolationBuilder->atPath('label')->willReturn($constraintViolationBuilder);
+        $constraintViolationBuilder->addViolation()->shouldBeCalled();
+
+        $this->validate($datagridView, $constraint)->shouldReturn(null);
```

Comment:
return is not important

```suggestion
        $this->validate($datagridView, $constraint);
```